### DISCUSSION TITLE
End Game concepts/ideation

### DISCUSSION THREAD

**Discussion by Volte6** [GoMudEngine/GoMud#117] [General]
# Motivation

We need to have an eye on what the long term gameplay loop will be, and how the end-game plays into that.

To prevent massive power spawn due to levelling, player stats (vitality, strength, smarts, etc) only grow linearly up to about 105, but then the additional stats start to apply logarithmically.
**Example**

Stat Value | Adjusted
-- | --
81 | 81
91 | 91
101 | 101
111 | 107
121 | 109
... | ...
200 | 120

## End Game

The end game is really how does a player "complete" the game? As a general concept MUD's often just have an open end - players persist forever and just keep amassing treasure, fortune and power.

This is fine, I don't want to take this from players, as it is actually a pleasant reality of MUD's.

That said, there should be a designed-for "farthest or greatest challenge" for players to overcome. 

Think: Final Boss, Final Quest, etc

## Considerations

* End game should not significantly change the settings, lore and content of the world.
  * E.g. `Frostfang` should not become a non-winter setting, `The King's Shadow` quest should not be removed.
    * Note: There may be an opportunity/way to cause areas/quests to swap, but ideally it would be designed in a manner that the next "end game" it swaps back, so no content is thrown away.
* End game should have no permanent negative ramifications for the player, unless they opt in to it or receive something in return. 
  * E.g. Rerolling their character

## End Game Conditions

Some possible conditions that represent an "end game"

* Reaching a certain level or higher
* Completing a specific "end game" quest
* Defeating a Big Boss

## Initial Ideas

**Reroll with a perk**

The player character rerolls but gets an upgrade unavailable to a normal newly created character:

* Choose from an `exclusive race` such as a player specific one ("`angel`"), or one of the mob races like `troll` or `insect` - each with their own special powers and limitations (challenging).
* Reroll but have a new `perma-buff` selected from a list - can repeat to stack more buffs
* Pick form one or more special `Items` - A set of Paragon gear they can pick one of each time. Maybe multiple sets to try and build up.
* Special `skills` that are normally untrainable (less likely to do this, since player skills make the world more interesting for everyone, but maybe something is too powerful)
* Special `spells` that are normally unlearnable.

**Reroll and BECOME the final boss**

When "The Final Boss" is defeated, it is replaced with the players character, who then becomes the Final Boss Mob. This has the added effect of increasing the difficult of the final boss with each player that gets to the end game.

**Gain Control of an Area or Feature**

When ever a new player reaches the end game, they become anointed into a special role, usurping and replacing the previous player who occupied that role (if any).
* Become the `governor` of an area, dictating certain policies or settings
  * i.e. Taxes, cost of player homes, guards aggression towards different alignments, the activation of certain events, etc.
  * The governor could naturally have their own house

**Comment by andrewgoat** [comment]
I have become interested in late with GoMUD and its progress. I originally saw it on reddit in /r/MUD. This is my first post on the Github Discussion for the codebase, and I find this particular topic fascinating.

Before I get to my main idea for Endgame, I would simply say that I am 41 years old, and I have played MUDs of different varieties off and on since about 1996.

One thing I have always daydreamed about building my own MUD around would be the idea of a 'Wish' system that goes in seasons. As in, players vying to be the one granted a Wish from the Gods or whatever ultimate power fits the theme of the game. Players could compete for this in any number of ways, but due to the uniqueness of it, I think it would be best to have one person per in game season win it. Seasons being like your typical, 3-4 real life month cycles that other games do.

The Wish itself could be derived from maybe a massive list of super powerful things, say like 50 different wishes, but the player would only be presented with 3 different ones from that pool, at random, perhaps skewed towards the alignment of their character. IE: Evil player gets to wish to do oppressive things, selfish things, etc.

My idea for this was to have like different player ran cities/nations vying for it across warfare against each other, or tournaments, things like that. I am sure this idea could be flexible, however.

Anyways, just a random thought. Love what you've done with GoMUD so far!
